core/src/tests/dummy_transport.rs

```diff
@@ -85,14 +91,31 @@ impl Transport for DummyTransport {
         }
     }
 
-    fn dial(self, _addr: Multiaddr) -> Result<Self::Dial, (Self, Multiaddr)>
+    fn dial(self, addr: Multiaddr) -> Result<Self::Dial, (Self, Multiaddr)>
     where
         Self: Sized,
     {
-        unimplemented!();
+        let peer_id = if let Some(peer_id) = self.next_peer_id {
+            peer_id
+        } else {
+            PublicKey::Rsa((0 .. 128).map(|_| -> u8 { 1 }).collect()).into_peer_id()
+        };
+
+        let fut =
+            if addr.to_string() == "/unix/unreachable" {
```

Comment:
Same here. This is extremely arbitrary.

Reply:
Here the context are these two tests: [one](https://github.com/libp2p/rust-libp2p/pull/602/files#diff-0c5110fdcca8841806bc220275bed002R1470), [two](https://github.com/libp2p/rust-libp2p/pull/602/files#diff-0c5110fdcca8841806bc220275bed002R1498) that tests that the appropriate errors are propagated. I needed a way to emulate an unreachable host.
What's a better way? I could perhaps make an "EvenDummerDummyTransport" with this hack in it. Or add a test-only "unreachable" protocol variant?

Reply:
I changed this to use the IPv6 "black hole" prefix, `0100/::64`. Not sure if that's any better though. :/